### PR TITLE
Fix/tabs keyboard onclick

### PR DESCRIPTION
**Pull request by minevala**
## Description
- handle Tab onClick also on enter and space keypresses

## Related Issue
- https://helsinkisolutionoffice.atlassian.net/browse/HDS-1239

## Motivation and Context
- Tab (without onClick) behaves similarly if the user clicks the mouse or presses enter or space key on Tab. But now only a mouse click triggers a Tab's custom onClick handler if it exists. This causes problems when onClick is used for custom navigation between tabs.

## How Has This Been Tested?
- Locally on dev machine

👉 [Demo](https://city-of-helsinki.github.io/hds-demo/tab-onclick-enter-space/iframe.html?id=components-tabs--with-custom-on-click-action&args=&viewMode=story)

### CODE REVIEW

File: packages/react/src/components/tabs/Tabs.test.tsx [hunk 1]
```diff
@@ -35,4 +37,36 @@ describe('<Tabs /> spec', () => {
     const results = await axe(container);
     expect(results).toHaveNoViolations();
   });
+
+  it('should call onClick when user clicks tab', () => {
```
Comment (VilleMiekkoja) [inline packages/react/src/components/tabs/Tabs.test.tsx:41]: Nice unit tests 👍

File: packages/react/src/components/tabs/Tabs.stories.tsx [hunk 2]
```diff
@@ -112,3 +112,46 @@ export const WithCustomTheme = () => {
...
+
+  const mockLoading = () => {
+    setIsLoading(true);
+    setTimeout(() => {
+      setIsLoading(false);
+    }, 200);
+  };
+
+  const onTabClick = (activeIndex) => {
+    setActiveTabIndex(activeIndex);
```
Comment (VilleMiekkoja) [inline packages/react/src/components/tabs/Tabs.stories.tsx:133]: Hmm.. I think there's internal logic in the Tab component where it sets the active index on tab click. So Im not sure if this is needed?

Comment (minevala) [reply packages/react/src/components/tabs/Tabs.stories.tsx]: -[x] remove unnecessary state update from story onClick

**Review by VilleMiekkoja — APPROVED** [review]
Nice fix 🏅

**Pull request merged**